pkg/metrics/metric/counter.go

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package metric
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+func NewCounter(opts CounterOpts) Counter {
+	return &counter{
+		Counter: prometheus.NewCounter(opts.toPrometheus()),
+		metric: metric{
+			enabled: opts.EnabledByDefault,
+			opts:    Opts(opts),
+		},
+	}
+}
+
+type Counter interface {
+	prometheus.Counter
+	WithMetadata
+
+	Get() float64
+}
+
+type counter struct {
+	prometheus.Counter
+	metric
+}
+
+func (c *counter) Describe(descChan chan<- *prometheus.Desc) {
```

Comment:
Could we have the underlying Counter type be composed with counter, that way we don't have to re implement these functions (and you can still override for Collect()).